ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableTest.java

```diff
@@ -122,56 +135,125 @@ public void shouldThrowIfGettingStateStoreFails() {
   }
 
   @Test
-  public void shouldThrowIfStoreGetFails() {
+  public void shouldKeyQueryWithCorrectParams() {
     // Given:
-    when(tableStore.get(any())).thenThrow(new MaterializationTimeOutException("Boom"));
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getOnlyPartitionResult()).thenReturn(keyResult);
+    when(keyResult.getResult()).thenReturn(row);
+    when(row.value()).thenReturn(ROW1);
```

Comment:
Man, this mocking stuff is brutal. Do you think it would be a good idea to add a utility method to create results like:

```
private static <R> StateQueryResult<ValueAndTimestamp<R> makeResult(final R result) {
        final StateQueryResult<ValueAndTimestamp<R>> result = new StateQueryResult<>();
        result.addResult(PARTITION, QueryResult.forResult(ValueAndTimestamp.make(result, -1)));
        return result;
}
```

And then be able to just do:
```suggestion
    when(kafkaStreams.query(any())).thenReturn(makeResult(ROW1));
```

The reason it might be nice is that if we change how the code interacts with the result, we're going to have to change every single test case as well. E.g., if you take the suggestion to get the specific partition instead of "getOnlyPartitionResult", or when we add logic to extract the Position later on. It's better if unit tests are generally able to ignore irrelevant interactions, since changing code AND tests is always riskier than just changing code and observing that all the tests still pass.

Just a suggestion, feel free to take it/leave it/leave it for future refactoring.

Reply:
This is a great suggestions, thank you!